0.6.1 is still broken for armv5te

0.6.1 is still broken for armv5te:

```
$ cargo build --target armv5te-unknown-linux-musleabi
Compiling zip v0.6.1
error[E0412]: cannot find type `AtomicU64` in module `atomic`
    --> /root/.cargo/registry/src/github.com-1ecc62[9](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:9)9db9ec823/zip-0.6.1/src/types.rs:258:30
     |
258  |   pub struct AtomicU64(atomic::AtomicU64);
     |                                ^^^^^^^^^ help: a struct with a similar name exists: `AtomicU[16](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:16)`
     |
note: struct `crate::write::zip_writer::AtomicU64` exists but is inaccessible
    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/zip-0.6.1/src/types.rs:258:1
     |
258  | pub struct AtomicU64(atomic::AtomicU64);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not accessible

error[E0433]: failed to resolve: could not find `AtomicU64` in `atomic`
    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/zip-0.6.1/src/types.rs:262:[22](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:22)
     |
262  |           Self(atomic::AtomicU64::new(v))
     |                        ^^^^^^^^^ help: a struct with a similar name exists: `AtomicU16`
     |
note: struct `crate::write::zip_writer::AtomicU64` exists but is inaccessible
    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec8[23](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:23)/zip-0.6.1/src/types.rs:[25](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:25)8:1
     |
258  | pub struct AtomicU64(atomic::AtomicU64);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not accessible

error[E0433]: failed to resolve: could not find `AtomicU64` in `atomic`
    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/zip-0.6.1/src/types.rs:[28](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:28)0:22
     |
280  |           Self(atomic::AtomicU64::new(self.load()))
     |                        ^^^^^^^^^ help: a struct with a similar name exists: `AtomicU16`
     |
note: struct `crate::write::zip_writer::AtomicU64` exists but is inaccessible
    --> /root/.cargo/registry/src/github.com-1ecc6[29](https://github.com/messense/aliyundrive-webdav/runs/5796858930?check_suite_focus=true#step:5:29)9db9ec823/zip-0.6.1/src/types.rs:258:1
     |
258  | pub struct AtomicU64(atomic::AtomicU64);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not accessible
```

_Originally posted by @messense in https://github.com/zip-rs/zip/issues/284#issuecomment-1086500368_

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ sha1 = {version = "0.10.1", optional = true }
 time = { version = "0.3.7", features = ["formatting", "macros" ], optional = true }
 zstd = { version = "0.10.0", optional = true }
 
-[target.'cfg(any(target_arch = "mips", target_arch = "powerpc"))'.dependencies]
+[target.'cfg(any(all(target_arch = "arm", target_pointer_width = "32"), target_arch = "mips", target_arch = "powerpc"))'.dependencies]
 crossbeam-utils = "0.8.8"
 
 [dev-dependencies]

--- a/src/types.rs
+++ b/src/types.rs
@@ -2,10 +2,18 @@
 #[cfg(doc)]
 use {crate::read::ZipFile, crate::write::FileOptions};
 
-#[cfg(not(any(target_arch = "mips", target_arch = "powerpc")))]
+#[cfg(not(any(
+    all(target_arch = "arm", target_pointer_width = "32"),
+    target_arch = "mips",
+    target_arch = "powerpc"
+)))]
 use std::sync::atomic;
 
-#[cfg(any(target_arch = "mips", target_arch = "powerpc"))]
+#[cfg(any(
+    all(target_arch = "arm", target_pointer_width = "32"),
+    target_arch = "mips",
+    target_arch = "powerpc"
+))]
 mod atomic {
     use crossbeam_utils::sync::ShardedLock;
     pub use std::sync::atomic::Ordering;

